### ISSUE TITLE
false error: "NewType cannot be used with a class created with NewType"

### ISSUE THREAD

**Issue by ghost** [microsoft/pyright#6976]
I started to create a discussion, but upon closer inspection I'm relatively sure this is a bug, since it contradicts the explicit documentation:

> However, it is possible to create a [NewType](https://docs.python.org/3/library/typing.html#newtype) based on a ‘derived’ NewType:
> 
> ```python
> from typing import NewType
> 
> UserId = NewType('UserId', int)
> 
> ProUserId = NewType('ProUserId', UserId)
> ```
> and typechecking for ProUserId will work as expected.

### Reproduction:

`paths.py`
```python
from __future__ import annotations

from pathlib import Path
from typing import NewType

Dir = NewType("Dir", Path)
ConfigDir = NewType("ConfigDir", Dir)

p = Path(".")
print(p)
if p.is_dir():
    d = Dir(p)
    print(d)

    if d / ".config":
        c = ConfigDir(d)
        print(c)
```

```console
$ python --version
Python 3.12.1

$ python paths.py
.
.
.

$ pip install pyright --upgrade
Requirement already satisfied: pyright in ./venv/lib/python3.12/site-packages (1.1.345)

$ pyright --version
pyright 1.1.345

$ pyright paths.py
./paths.py
  ./paths.py:7:34 - error: NewType cannot be used with a class created with NewType
1 error, 0 warnings, 0 informations
exit code: 1
```

### Background

Starting at version 1.1.343, I'm getting this new error on previously-clean code but I can't see a reason for it. What's invalid about this code? I don't see anything in the NewType docs that suggests this wouldn't work just fine, and it does seem to. At runtime it (of course) works just fine, but also I found that pyright (prior to 1.1.343) was helping me find bugs earlier than I would have otherwise with this typing scheme.

**Comment by bukzor-sentryio** [comment]
secondarily, I'm not finding a way to suppress this diagnostic in the meantime. As I understand it, this should have done the trick, but seems to have no effect:


```
# pyright: reportGeneralTypeIssues=false
Dir = NewType("Dir", Path)
ConfigDir = NewType("ConfigDir", Dir)
# pyright: reportGeneralTypeIssues=true
```

```console
$ pyright paths.py --outputjson
{
    "version": "1.1.345",
    "generalDiagnostics": [
        {   "severity": "error",
            "message": "NewType cannot be used with a class created with NewType",
        }
    ],
}
```

**Comment by erictraut** [comment]
Thanks for reporting this. I misread the documentation somehow and thought that it required this to be flagged as an error. I'll revert that change.

As for suppressing the diagnostic, the `# pyright: reportGeneralTypeIssues=<X>` is a file-level directive, so you shouldn't use more than one of them in the same file — especially if they are contradictory. Second, this diagnostic isn't associated with the `reportGeneralTypeIsssues` rule, so this directive won't have any affect on this diagnostic anyway. You would need to use a `# type: ignore` or `# pyright: ignore` comment on that line if you want to suppress it. Alternatively, you can pin to an earlier version of pyright.

I'll prioritize a fix so this is addressed in the next release.

**Comment by bukzor-sentryio** [comment]
I'm happy to hear that my bug reports are getting more accurate :)
Thank you so much for your diligence and your time.

**Comment by erictraut** [comment]
For reference, this was a regression introduced as part of [this issue](https://github.com/microsoft/pyright/issues/6830).

It will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.346, which I just published. It will also be included in a future release of pylance.
